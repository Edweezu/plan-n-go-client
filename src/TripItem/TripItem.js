import React from 'react'
import TripsContext from '../TripsContext'
import { Link } from 'react-router-dom'
// import Trip from '../Trip/Trip';
import config from '../config'
import TokenService from '../services/token-service'
import { format } from 'date-fns'
import { confirmAlert } from 'react-confirm-alert'
import 'react-confirm-alert/src/react-confirm-alert.css'

class TripItem extends React.Component {
    static contextType = TripsContext

    state = {
        error: null
    }

    handleDeleteForm = () => {
        confirmAlert({
            title: '',
            message: 'Are you sure you want to delete this?',
            buttons: [
                {
                    label: 'Yes',
                    onClick : () => {
                        this.handleClickDelete()
                    }
                },
                {
                    label: 'No',
                    onClick : () => {
                        this.setState({
                            error: null
                        })
                    }
                }
            ]
        })
    }
    

    handleClickDelete = (e) => {

        const tripid = this.props.id

        return fetch(`${config.API_ENDPOINT}/trips/${tripid}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
        })
        .then(() => {
            this.context.deleteTrip(tripid)
        })
        .catch(error => {
            this.setState({
                error: error
            })
        })
    }

    formatDate = (startdate='', enddate='') => {
        let [startYear, startMonth, startDay ] = startdate.substr(0, 10).split('-')
        let [endYear, endMonth, endDay ] = enddate.substr(0, 10).split('-')
        if (startMonth !== endMonth) {
            return format(new Date(startYear, (startMonth - 1), startDay), 'MMM D') + ' - ' + format(new Date(endYear, (endMonth - 1), endDay), 'MMM D, YYYY')
        } else {
            return format(new Date(startYear, (startMonth - 1), startDay), 'MMM D') + ' - ' + format(new Date(endYear, (endMonth - 1), endDay), 'D, YYYY')
        }
    }

    render () {
        const { id, name, startDate, endDate} = this.props

        return (
            <section className='trip-item'>
                <h2 className='trip-item-title'>
                    <Link to={"/trip/" + id}>
                        {name}
                    </Link>
                    <i onClick={this.handleDeleteForm} className="far fa-trash-alt"></i>
                </h2>
                <p>
                    {this.formatDate(startDate, endDate)}
                </p>
            </section>
        )
    }
}

export default TripItem