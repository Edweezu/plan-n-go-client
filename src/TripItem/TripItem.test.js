import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import TripItem from './TripItem'


describe('TripItem component', () => {
    const props = {
        tripid: 1,
        name: 'Seattle',
        startDate: '2019-07-17T00:00:00.000Z',
        endDate: '2019-07-25T00:00:00.000Z'
    }

    it('renders the given component', () => {
        const wrapper = shallow(<TripItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<TripItem {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})