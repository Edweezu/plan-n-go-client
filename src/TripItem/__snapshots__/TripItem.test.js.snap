// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripItem component renders the component's given props 1`] = `
<section
  className="trip-item"
>
  <h2
    className="trip-item-title"
  >
    <Link
      to="/trip/undefined"
    >
      Seattle
    </Link>
    <i
      className="far fa-trash-alt"
      onClick={[Function]}
    />
  </h2>
  <p>
    Jul 17 - 25, 2019
  </p>
</section>
`;

exports[`TripItem component renders the given component 1`] = `
<section
  className="trip-item"
>
  <h2
    className="trip-item-title"
  >
    <Link
      to="/trip/undefined"
    />
    <i
      className="far fa-trash-alt"
      onClick={[Function]}
    />
  </h2>
  <p>
    Invalid Date - Invalid Date
  </p>
</section>
`;
